search-provider/yelp-search-provider.c

```diff
@@ -0,0 +1,568 @@
+                                              terms,
+                                              get_search_results);
+    g_idle_add_full (G_PRIORITY_DEFAULT_IDLE,
+                     check_page_data,
+                     delayed,
+                     (GDestroyNotify) delayed_initial_result_set_free);
+    return TRUE;
+}
+
+static gboolean
+handle_get_subsearch_result_set (YelpShellSearchProvider2 *skeleton,
+                                 GDBusMethodInvocation    *invocation,
+                                 gchar                   **previous_results,
+                                 gchar                   **terms,
+                                 gpointer                  user_data)
+{
+    DelayedInitialResultSet *delayed;
+    YelpSearchProviderApp *app = YELP_SEARCH_PROVIDER_APP (user_data);
+
+    if (g_strv_length (terms) < 2 && g_utf8_strlen (terms[0], -1) < 3 ) {
+        g_dbus_method_invocation_return_value (invocation, g_variant_new ("(as)", NULL));
+        return TRUE;
+    }
+
+    if (g_hash_table_size (app->page_data_hash_map) > 0) {
+        g_dbus_method_invocation_return_value (invocation,
+                                               get_search_results (terms,
+                                                                   app->page_data_hash_map));
+        return TRUE;
+    }
+
+    delayed = delayed_initial_result_set_new (app,
+                                              invocation,
+                                              terms,
+                                              get_search_results);
+    g_idle_add_full (G_PRIORITY_DEFAULT_IDLE,
+                     check_page_data,
+                     delayed,
+                     (GDestroyNotify) delayed_initial_result_set_free);
+    return TRUE;
+}
+
+static gboolean
+handle_get_result_metas (YelpShellSearchProvider2  *skeleton,
+                         GDBusMethodInvocation     *invocation,
+                         gchar                    **results,
+                         gpointer                   user_data)
+{
+
+    DelayedInitialResultSet *delayed;
+    YelpSearchProviderApp *app = YELP_SEARCH_PROVIDER_APP (user_data);
+
+    if (g_hash_table_size (app->page_data_hash_map) > 0) {
+        g_dbus_method_invocation_return_value (invocation,
+                                               get_result_metas (results,
+                                                                 app->page_data_hash_map));
+        return TRUE;
+    }
+
+    delayed = delayed_initial_result_set_new (app,
+                                              invocation,
+                                              results,
+                                              get_result_metas);
+    g_idle_add_full (G_PRIORITY_DEFAULT_IDLE,
+                     check_page_data,
+                     delayed,
+                     (GDestroyNotify) delayed_initial_result_set_free);
+    return TRUE;
+}
+
+static gboolean
```

Comment:
Yeah, github sometimes moves the comment elsewhere, to the fitting location. It did it to me too.